src/cheshire/generate.clj

```diff
@@ -43,7 +44,9 @@
                                      ~g ~(with-meta o {:tag `BigDecimal}))
                          Ratio (.writeNumber ~g (double ~o))
                          Short (.writeNumber ~g (int ~o))
-                         Byte (.writeNumber ~g (int ~o))]]
+                         Byte (.writeNumber ~g (int ~o))
+                         AtomicLong (.writeNumber ~g (long ~o))
+                         AtomicInteger (.writeNumber ~g (int ~o))]]
```

Comment:
So looking at Clojure's `int` implementation, it appears that it will first cast it to a long, then an int. I wonder if the performance would be better if instead we did something somewhat like:


```suggestion
                         AtomicLong (.writeNumber ~g (.get ~o))
                         AtomicInteger (.writeNumber ~g (.get ~o))]]
```

to avoid Clojure's unwrapping? I think I may take a stab at benchmarking this to see if it makes any difference here.